src/main/java/gregtech/api/recipes/category/GTRecipeCategory.java

```diff
@@ -0,0 +1,104 @@
+package gregtech.api.recipes.category;
+
+import gregtech.api.recipes.RecipeMap;
+import it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Map;
+
+public final class GTRecipeCategory {
+
+    private static final Map<String, GTRecipeCategory> categories = new Object2ObjectOpenHashMap<>();
+
+    private final String modid;
+    private final String name;
+    private final String uniqueID;
+    private final String unlocalizedName;
+    private final RecipeMap<?> recipeMap;
+    private Object icon;
+
+    /**
+     * Create a GTRecipeCategory
+     *
+     * @param modid the mod id of the category
+     * @param name the name of the category, used in translation. Uses the recipeMap's translation key if null.
+     * @param recipeMap the recipemap that accepts this category
+     * @return the new category
+     */
+    @Nonnull
+    public static GTRecipeCategory create(@Nonnull String modid, @Nullable String name, @Nonnull RecipeMap<?> recipeMap) {
+        String key = name == null ? recipeMap.getUnlocalizedName() : name;
```

Comment:
I think that the name should just be nonnull here, as all times this category is created, access to the recipemap is available, so the recipemap name can be retrieved.